src/editor.js

```diff
@@ -254,23 +262,26 @@ const FileUploadButton = () => html`
  * @param {Promise<Element>=} data.codeMirrorElement
```

Comment:
Should we start using some type of struct for these typedefs that keep growing?

Reply:
Not sure. Maybe for type safety it would make sense but I believe this jsdoc format is also statically analyzable.

For conciseness, this is shorter, since it's inline and most template functions take unique state slices 😉